File: Coding_Questions/C++/FindDuplicateValue.cpp
// ---Find Duplicate Value---

#include <vector>
#include <set>


int find_duplicate_value(std::vector<int> array) {
    std::set<int> holding_set;

    for (const auto &element : array) {
        if (holding_set.find(element) != holding_set.cend()) {
            return element;
        }
        holding_set.insert(element);
    }

    return -1;
}

File: Coding_Questions/JavaScript/FindDuplicateValue.js
// ---Find Duplicate Value---

const firstDuplicateValue = (array) => {
  const holdingSet = new Set();

  for (const element of array) {
    if (holdingSet.has(element)) {
      return element;
    }
    holdingSet.add(element);
  }

  return -1;
};

const input = [2, 1, 5, 2, 3, 3, 4];
const actual = firstDuplicateValue(input);
console.log(actual); // 2
